File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/TestReport.ts
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type TestReportParticipant = BackboneElement & {
    uri: string;
    type: string;
    display?: string;
};

export type TestReportSetupActionAssert = BackboneElement & {
    detail?: string;
    result: string;
    message?: string;
};

export type TestReportSetupActionOperation = BackboneElement & {
    detail?: string;
    result: string;
    message?: string;
};

export type TestReportSetupAction = BackboneElement & {
    assert_?: TestReportSetupActionAssert;
    operation?: TestReportSetupActionOperation;
};

export type TestReportSetup = BackboneElement & {
    action: TestReportSetupAction[];
};

export type TestReportTeardownAction = BackboneElement & {
    operation: Reference;
};

export type TestReportTeardown = BackboneElement & {
    action: TestReportTeardownAction[];
};

export type TestReportTestAction = BackboneElement & {
    assert_?: Reference;
    operation?: Reference;
};

export type TestReportTest = BackboneElement & {
    name?: string;
    action: TestReportTestAction[];
    description?: string;
};

export type TestReport = DomainResource & {
    tester?: string;
    name?: string;
    testScript: Reference;
    participant?: TestReportParticipant[];
    setup?: TestReportSetup;
    status: string;
    result: string;
    score?: number;
    identifier?: Identifier;
    issued?: string;
    teardown?: TestReportTeardown;
    test?: TestReportTest[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/TestReport.py
from typing import Optional, List
from pydantic import *
from ..base import *

class TestReport_Participant(BackboneElement):
    uri: str
    type: str
    display: Optional[str] = None

class TestReport_Setup_Action_Assert(BackboneElement):
    detail: Optional[str] = None
    result: str
    message: Optional[str] = None

class TestReport_Setup_Action_Operation(BackboneElement):
    detail: Optional[str] = None
    result: str
    message: Optional[str] = None

class TestReport_Setup_Action(BackboneElement):
    assert: Optional[TestReport_Setup_Action_Assert_] = None
    operation: Optional[TestReport_Setup_Action_Operation] = None

class TestReport_Setup(BackboneElement):
    action: list[TestReport_Setup_Action] = []

class TestReport_Teardown_Action(BackboneElement):
    operation: Reference

class TestReport_Teardown(BackboneElement):
    action: list[TestReport_Teardown_Action] = []

class TestReport_Test_Action(BackboneElement):
    assert: Optional[Reference] = None
    operation: Optional[Reference] = None

class TestReport_Test(BackboneElement):
    name: Optional[str] = None
    action: list[TestReport_Test_Action] = []
    description: Optional[str] = None

class TestReport(DomainResource):
    tester: Optional[str] = None
    name: Optional[str] = None
    test_script: Reference
    participant: Optional[List[TestReport_Participant]] = None
    setup: Optional[TestReport_Setup] = None
    status: str
    result: str
    score: Optional[float] = None
    identifier: Optional[Identifier] = None
    issued: Optional[str] = None
    teardown: Optional[TestReport_Teardown] = None
    test: Optional[List[TestReport_Test]] = None